File: POOBanca/Java/Jornal.java
package poobanca;

public class Jornal extends Produto {
    protected boolean diario;
    
    @Override
    public void apresentacao() {
        super.apresentacao();
        
        System.out.println("O jornal é diário? " + (this.diario ? "Sim" : "Não") + ".");
    }
    
    // Construtor
    public Jornal(String nome, String editora, int totPaginas, boolean diario) {
        super(nome, editora, totPaginas);
        
        this.diario = diario;
    }
    
    // Getter e Setter
    public boolean getDiario() {
        return this.diario;
    }
    
    public void setDiario(boolean diario) {
        this.diario = diario;
    }
}

File: POOBanca/PHP/Jornal.php
<?php
require_once "Produto.php";

class Jornal extends Produto {
    protected $diario; // Boolean

    // Override
    public function apresentacao() {
        parent::apresentacao();

        echo "O jornal é diário? " . ($this->diario ? "Sim" : "Não") . ".<br/>";
    }

    // Construtor e Destrutor
    public function __construct($nome, $editora, $totPaginas, $diario) {
        parent::__construct($nome, $editora, $totPaginas);

        $this->diario = $diario;
    }

    public function __destruct() {
        echo "<p>Objeto Jornal Destruído com Sucesso!</p>";
    }

    // Getter e Setter
    public function getDiario() {
        return $this->diario;
    }

    public function setDiario($diario) {
        $this->diario = $diario;
    }
}